Makefile

```diff
@@ -334,6 +334,9 @@ e2e-gh-token: build-scorecard check-env | $(GINKGO)
 	# Run e2e tests. GITHUB_AUTH_TOKEN set to secrets.GITHUB_TOKEN must be used to run this.
 	TOKEN_TYPE="GITHUB_TOKEN" $(GINKGO) --race -p -v -cover -coverprofile=e2e-coverage.out --keep-separate-coverprofiles ./...
 
+e2e-gitlab: ## Runs e2e tests for GitLab only. TOKEN_TYPE is arbitrary, but must be set to something
+	TOKEN_TYPE="GITLAB_PAT" $(GINKGO) --race -p -vv --focus '.*GitLab' ./...
```

Comment:
-> Why do we need `TOKEN_TYPE` for GitLab?
-> Can we run `./clients/gitlabrepo/...` and remove `--focus '.*GitLab'`?

Reply:
it may prevent the gitlab tests from running during `make e2e-pat` or `make e2e-gh-token` ?

Reply:
Correct, IIUC `e2e-pat` and `e2e-gh-token` should be running the GitLab tests already give scope of `./...`. So either we remove the extra `e2e-gitlab` or narrow the scopes for it to only run on GitLab.

Reply:
Updated this:
* e2e-gitlab-token will be a make target that can be used from the CLI. The user can pass in their GitLab token to run this test.
* e2e-gitlab will be a make target for unauthenticated tests that can run from CI. The rate limit is 500 requests/minute, which we should be well under for now. If this is an issue, it can be split into its own workflow file and marked optional.
* We can't just run ./clients/gitlabrepo since some future tests will be in `e2e/`